### PR TITLE
Add trust to ad domain 5731

### PR DESCRIPTION
**Pull request by venkykuberan**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
AD Domain - added new resource `google_active_directory_domain_trust` 
```

### CODE REVIEW

File: templates/terraform/custom_delete/active_directory_domain_trust.go.erb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+    config := meta.(*Config)
```
Comment (ndmckinley) [inline templates/terraform/custom_delete/active_directory_domain_trust.go.erb:1]: Hm, what is different from the generated code here?  A comment might help make it clearer why this is being done.

Comment (venkykuberan) [reply templates/terraform/custom_delete/active_directory_domain_trust.go.erb:1]: The generate DELETE method isn't including the {trust: } object in the response body thus custom_delete is needed

Comment (ndmckinley) [reply templates/terraform/custom_delete/active_directory_domain_trust.go.erb:1]: Hm.  I wish we could solve that in a more targeted way.  Let me check with Dana and see if we're okay to do it this way or if we need a more complicated (but easier to maintain) change.

Comment (ndmckinley) [reply templates/terraform/custom_delete/active_directory_domain_trust.go.erb:1]: Okay, this is fine!  Can you make a big noisy comment in api.yaml and terraform.yaml that says that this resource's delete function needs to be modified any time we edit the schema?  After that, as long as the tests are passing and all seems to be working, I'm good to approve.

**Pull request merged**